### PR TITLE
feat: add mongodb support

### PR DESCRIPTION
**Pull request by LewdHuTao**
This PR will allow user to choose between mongodb or postgresql to use on their bot

### CODE REVIEW

**Comment by LewdHuTao** [comment]
known issue:
playlist schema for mongodb not working properly

error:
```
Promise {
  <rejected> PrismaClientKnownRequestError:
  Invalid `client.db.playlist.create()` invocation in
  C:\Users\Discord-MusicBot-feature-DJSv14\djs-bot\commands\music\playlists.js:153:46

    150 if (!playlistName) return interaction.reply({ embeds: [new EmbedBuilder().setColor(client.config.embedColor).setDescription("You need to provide a name for the playlist")], ephemeral: true });
    151 else if (playlistName.length < 3) return interaction.reply({ embeds: [new EmbedBuilder().setColor(client.config.embedColor).setDescription("The playlist name can't be shorter than 3 characters")], ephemeral: true });
    152 else if (playlistName.length > 32) return interaction.reply({ embeds: [new EmbedBuilder().setColor(client.config.embedColor).setDescription("The playlist name can't be longer than 32 characters")], ephemeral: true });
  → 153 const playlist = await client.db.playlist.create(
  Inconsistent column data: Malformed ObjectID: provided hex string representation must be exactly 12 bytes, instead got: "454278022132924417", length 18.    
      at vn.handleRequestError (C:\Users\Iman\Downloads\djsv14\Discord-MusicBot-feature-DJSv14\djs-bot\node_modules\@prisma\client\runtime\library.js:123:6730)
      at vn.handleAndLogRequestError (C:\Users\Iman\Downloads\djsv14\Discord-MusicBot-feature-DJSv14\djs-bot\node_modules\@prisma\client\runtime\library.js:123:6119)
      at vn.request (C:\Users\Iman\Downloads\djsv14\Discord-MusicBot-feature-DJSv14\djs-bot\node_modules\@prisma\client\runtime\library.js:123:5839)
      at async l (C:\Users\Iman\Downloads\djsv14\Discord-MusicBot-feature-DJSv14\djs-bot\node_modules\@prisma\client\runtime\library.js:128:9763)
      at async SlashCommand.run (C:\Users\Iman\Downloads\djsv14\Discord-MusicBot-feature-DJSv14\djs-bot\commands\music\playlists.js:153:21) {
    code: 'P2023',
    clientVersion: '5.2.0',
    meta: {
      message: 'Malformed ObjectID: provided hex string representation must be exactly 12 bytes, instead got: "454278022132924417", length 18.'
    }
  }
}
````

**Comment by LewdHuTao** [comment]
Everything is now ready except for mongodb playlist schema. So here how this thing work:

1. Choose db option in config or .env
2. Run the bot (`npm run db-start`)
3. DBScript will copy db schema (`Example: postgresql.prisma`) and paste it in default schema file (`schema.prisma`)

**Comment by LewdHuTao** [comment]
@BioCla now that all code are updated to djs v14 format. should we keep the 'lib/Embed.js' file?

image

**Comment by BioCla** [comment]
> @BioCla now that all code are updated to djs v14 format. should we keep the 'lib/Embed.js' file?
> 
> image

I think It's not needed anymore, but move it to the assets folder for reference

**Comment by BioCla** [comment]
> Everything is now ready except for mongodb playlist schema. So here how this thing work:
> 
> 1. Choose db option in config or .env
> 2. Run the bot (`npm run db-start`)
> 3. DBScript will copy db schema (`Example: postgresql.prisma`) and paste it in default schema file (`schema.prisma`)

Does it Have to copy the entire file? isnt there an option to select the prisma file from npx cl?

**Comment by LewdHuTao** [comment]
> Does it Have to copy the entire file? isnt there an option to select the prisma file from npx cl?

For now copy the entire file is the only way i found. Since the default schema file for prisma is `schema.prisma`, I tried to change the default file to `mongodb.schema/postgresql.schema` using this script `npx prisma generate --schema=./prisma/mongodb.schema` but it still need the `schema.prisma` file

File: djs-bot/lib/DBScript.js
Comment (BioCla) [inline djs-bot/lib/DBScript.js:1]: move to `./djs-bot/scripts/`

**Comment by BioCla** [comment]
> > Does it Have to copy the entire file? isnt there an option to select the prisma file from npx cl?
> 
> For now copy the entire file is the only way i found. Since the default schema file for prisma is `schema.prisma`, I tried to change the default file to `mongodb.schema/postgresql.schema` using this script `npx prisma generate --schema=./prisma/mongodb.schema` but it still need the `schema.prisma` file

alright, if everything works It's good 👍 . ready to merge whenever once all reviews are met

**Review by brianferri — CHANGES_REQUESTED** [review]

**Review by brianferri — APPROVED** [review]

**Pull request merged**